docs/RingBuffer.md

```diff
@@ -19,7 +19,30 @@ Mapped memory consumer:
 3. Directly read records from the producer pages (and update consumer offset as we read).
 4. Call `WaitForSingleObject`/`WaitForMultipleObject` as needed to wait for new data to be available.
 
-## API Changes
+### Differences from linux API
+
+Linux has only polling ring buffer consumers, even when callbacks are used.
+On linux the user code can call `ring_buffer__consume()` to invoke the callback on all available data,
+and `ring_buffer__poll()` to wait for data if needed and then consume available data.
+Linux consumers can also directly read from the mapped memory by using `mmap()` to map the data
+into user space and `ring_buffer__epoll_fd()` to get an epoll wait handle.
+
+On Windows asynchronous events are supported by default,
+so nothing extra needs to be done for callbacks to be invoked.
+
+If the `RINGBUF_FLAG_NO_AUTO_CALLBACK` flag is set, callbacks will not automatically be called and `ring_buffer__poll()`
+should be called to poll for available data and invoke the callback. On Windows a timeout of zero can be passed to
+`ring_buffer__poll()` to get the same behaviour as `ring_buffer__consume()`.
```

Comment:
What is the proposed behavior if `ring_buffer__consume()` is called when the `RINGBUF_FLAG_NO_AUTO_CALLBACK` flag is set?